azurerm/internal/services/network/tests/resource_arm_express_route_gateway_test.go

```diff
@@ -0,0 +1,238 @@
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMExpressRouteGateway_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_express_route_gateway", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMExpressRouteGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMExpressRouteGateway_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMExpressRouteGatewayExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMExpressRouteGateway_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_express_route_gateway", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMExpressRouteGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMExpressRouteGateway_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMExpressRouteGatewayExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMExpressRouteGateway_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_express_route_gateway"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMExpressRouteGateway_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_express_route_gateway", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMExpressRouteGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMExpressRouteGateway_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMExpressRouteGatewayExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMExpressRouteGateway_updateScaleUnits(data, 2),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMExpressRouteGatewayExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "scale_units", "2"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMExpressRouteGateway_tags(t *testing.T) {
```

Comment:
We can merge this and update test configs into one called complete, and then the update test can be basic -> complete -> basic

Reply:
I've made the changes as requested - but note that provisioning/updating an ExpressRoute gateway is a lengthy process (30mins-60mins in my experience), so this is going to be a long test (just timed out a 120m limit myself on this).